packages/testcontainers/src/container-runtime/clients/container/docker-container-client.ts

```diff
@@ -200,7 +211,7 @@ export class DockerContainerClient implements ContainerClient {
       stream.destroy();
 
       const inspectResult = await exec.inspect();
-      const exitCode = inspectResult.ExitCode === null ? -1 : inspectResult.ExitCode;
+      const exitCode = inspectResult.ExitCode ?? -1;
```

Comment:
I'm not familiar with Node idioms, but I've found the docs for [this operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing). IIUC, if the code is null or undefined, it will return -1, else, the code. And it makes sense to convert the ternary expression to the coalescence one. My question is about the usage across the library, is it used elsewhere so that we can accept the pattern? If not, I'd keep the PR without it and probably submit a separate one updating all the ternary usages if possible 🙏 Does it make sense to you?

Reply:
Honestly, this change solves a _SonarLint_ warning. It was an opportunistic change. As I was touching this file, I decided to solve this minor lint issue.

From my experience, performing small maintenance changes as you move along the codebase is better than making a big PR with multiple changes like this. This is mainly because, often, these changes are very _context-dependent_, and when you are working in a specific part of your codebase, you already have the context.

Said that, I don't have any strong opinion here; I would be okay reverting this line's changes 😁

Reply:
I'm fine adding it, not opposed to it at all. Just coming as a reviewer from another language makes it very difficult to adapt the mental model for new idioms very quickly, that's why I ask first to trust the author :)

For context, I'm not opposed to introduce anything new (as this one) given the tiny size of it. I just want to make sure we are clear and consistent for the sake of the project health, not introducing patterns "because yes".

In this case, I've double-checked the codebase, and it's already used 👏 So good to me! So thank you so much for your explanation about the rationale, as it makes me realise it'd be safe adding it even though not used before.

```
$ grep -R "??" .
./packages/testcontainers/src/container-runtime/clients/compose/default-compose-options.ts:  const log = options.logger ?? composeLog;
./packages/testcontainers/src/container-runtime/clients/image/docker-image-client.ts:      const authconfig = await getAuthConfig(imageName.registry ?? this.indexServerAddress);
./packages/testcontainers/src/container-runtime/clients/container/docker-container-client.ts:        since: opts?.since ?? 0,
./packages/testcontainers/src/container-runtime/auth/auths.ts:    const authEntries = dockerConfig.auths ?? {};
./packages/testcontainers/src/container-runtime/utils/pull-image.ts:    const authconfig = await getAuthConfig(options.imageName.registry ?? indexServerAddress);
./packages/testcontainers/src/utils/dockerfile-parser.ts:      .map((line) => line.replace(buildArgRegex, (_, arg) => buildArgs[arg] ?? ""))
./packages/testcontainers/src/generic-container/generic-container-builder.ts:        const authConfig = await getAuthConfig(imageName.registry ?? indexServerAddress);
./packages/testcontainers/src/generic-container/generic-container.ts:        ...(this.hostConfig.ExtraHosts ?? []),
./packages/testcontainers/src/generic-container/generic-container.ts:      ...(this.createOpts.Env ?? []),
./packages/testcontainers/src/generic-container/generic-container.ts:      ...(this.hostConfig.Ulimits ?? []),
./packages/testcontainers/src/generic-container/generic-container.ts:    this.hostConfig.CapAdd = [...(this.hostConfig.CapAdd ?? []), ...capabilities];
./packages/testcontainers/src/generic-container/generic-container.ts:    this.hostConfig.CapDrop = [...(this.hostConfig.CapDrop ?? []), ...capabilities];
./packages/testcontainers/src/generic-container/generic-container.ts:      ...(this.hostConfig.ExtraHosts ?? []),
./packages/testcontainers/src/reaper/reaper.ts:export const REAPER_IMAGE = process.env["RYUK_CONTAINER_IMAGE"] ?? "testcontainers/ryuk:0.5.1";
./packages/testcontainers/src/reaper/reaper.ts:    sessionId = reaperContainer?.Labels["org.testcontainers.session-id"] ?? new RandomUuid().nextUuid();
./packages/testcontainers/src/port-forwarder/port-forwarder.ts:export const SSHD_IMAGE = process.env["SSHD_CONTAINER_IMAGE"] ?? "testcontainers/sshd:1.1.0";
```